packages/cli/src/controllers/mfa.controller.ts

```diff
@@ -1,11 +1,21 @@
 import { Get, Post, RestController } from '@/decorators';
 import { BadRequestError } from '@/errors/response-errors/bad-request.error';
+import { ExternalHooks } from '@/external-hooks';
 import { MfaService } from '@/mfa/mfa.service';
 import { AuthenticatedRequest, MFA } from '@/requests';
 
 @RestController('/mfa')
 export class MFAController {
-	constructor(private mfaService: MfaService) {}
+	constructor(
+		private mfaService: MfaService,
+		private externalHooks: ExternalHooks,
+	) {}
+
+	@Post('/can-enable')
+	async canEnableMFA(req: AuthenticatedRequest) {
+		await this.externalHooks.run('mfa.beforeSetup', [req.user]);
+		return;
+	}
```

Comment:
This looks very odd that we delegate a check like this to an external hook. But I guess it is what it is

Reply:
it sucks. But the alternative it's doing it in the endpoint that actually enables MFA, but then the user will get the message after all steps are done, which is probably very annoying.

Reply:
Yes, agreed! Sadly this is check that only happens in cloud. I see how self hosted users will benefit from this as well, but we do not have the concept of email verification in n8n itself.